File: 3-Patron/php/Mapa.php
<?php
include './IMapa.php';
include './Mazmorra.php';
include './Textura.php';
include './Tesoro.php';

class Mapa implements IMapa{
	private $mazmorra;
	private $tesoro;
	private $textura;


	public function __construct($mazmorra, $tesoro, $textura){
    $this->mazmorra = $mazmorra;
    $this->tesoro = $tesoro;
    $this->textura = $textura;
	}

	public function getMazmorra(){
		return $this->mazmorra;	
	}

	public function setMazmorra($mazmorra){
		$this->mazmorra = $mazmorra;
	}

	public function getTesoro(){
		return $this->tesoro;	
	}

	public function setTesoro($tesoro){
		$this->tesoro = $tesoro;
	}

	public function getTextura(){
		return $this->textura;	
	}

	public function setTextura($textura){
		$this->textura = $textura;
	}

	public function verMapa(){
    return  "Mazmorra: " . $this->mazmorra .
            "Tesoro: " . $this->tesoro .
            "Textura: " $this->textura;
  	}
}
?>

File: 3-Patron/Java/Mapa.java
package buildercomposite;

public class Mapa implements IMapa{
    private Mazmorra mazmorra;
    private Tesoro tesoro;
    private Textura textura;
    
    public Mapa(Mazmorra mazmorra,Tesoro tesoro, Textura textura){
        this.mazmorra = mazmorra;
        this.tesoro = tesoro;
        this.textura = textura;
    }

    public Mazmorra getMazmorra(){
        return this.mazmorra;
    }
    public void setMazmorra(Mazmorra mazmorra){
        this.mazmorra = mazmorra;
    }
    public Tesoro getTesoro(Tesoro tesoro){
        return this.tesoro;
    }
    public void setTesoro(Tesoro tesoro){
        this.tesoro = tesoro;
    }
    public Textura getTextura(){
        return this.textura;
    }
    public void setTextura(Textura textura){
        this.textura = textura;
    }
    public String verMapa(){
        return ("Masmorra: "+  this.mazmorra+"\n Tesoro: "+this.tesoro+"\n Textura: "+this.textura);
    }
}
